### PR TITLE
Course Rollover

### LINKED ISSUE #686: Add Course Rollover function
A course may be rolled over into any educational (academic) year that includes dates after its start date. It may be rolled over with or without calendared events (offerings).

### PR DESCRIPTION
**Pull request by jrjohnson**
Added option to rollover a course.  User has to select a year and can optionally select a new start date and to skip offerings.

Fixes #686

Requires ilios/ilios#1475

todo:
- [x] Decide what should happen after a course is rolled over?  Right now it transitions to that course with a success message.

### CODE REVIEW

**Comment by jrjohnson** [comment]
@saschaben while saving a rollover requires API changes the UI here is ready for review and comments.

**Comment by saschaben** [comment]
rollover link in past would only display when course is in published /scheduled state, not in draft. we haven't done that in the current model.  thoughts?

**Comment by saschaben** [comment]
also: without the API update, I can't see the workflow. What happens as a user when I hit the go button? to I go to the new course? stay on the parent course? what is the success message? is there any pre-flight notice ("you are about to roll over course XXX into year YYY with all [values eg offerings] in place / removed...")...?

**Comment by saschaben** [comment]
generally this looks fine for a first run. 👍

**Comment by jrjohnson** [comment]
Workflow is you go to the new course summary page with a green notice of success.

**Comment by saschaben** [comment]
I'm good with that workflow.

let's add the preflight text.

**Comment by jrjohnson** [comment]
@stopfstedt this can be merged once ilios/ilios#1475 is in.  The tag is just a reminder of that.

File: app/router.js [hunk 1]
```diff
@@ -57,6 +58,7 @@ Router.map(function() {
   this.route('school', { path: 'schools/:school_id'});
   this.route('assign-students', {path: '/admin/assignstudents'});
   this.route('myprofile');
+  this.route('course-rollover');
```
Comment (stopfstedt) [inline app/router.js]: seems unnecessary. remove if possible, please.

**Comment by stopfstedt** [comment]
LGTM. will remove unnecessary route in separate PR.

**Pull request merged**